compiler/annotations_utils/util_parsing.py

```diff
@@ -66,8 +66,10 @@ def parse_arg_list_to_command_invocation(
     command, flags_options_operands
 ) -> CommandInvocationInitial:
     cmd_name = format_arg_chars(command)
+    #strip quotes from command where necessary
+    #quoted commands are interpreted as non-parallelizable
+    cmd_name = cmd_name.strip().strip('"').strip("'")
```

Comment:
I think it is better to do this change by making a new function replacing `format_arg_chars` since that function has access to the actual character codes and not just the string, making any transformation more robust! At this point in the execution, it should certainly be the case that `command` only is expanded and only has quotes (`Q` characters), normal (`C`), and escaped (`E`) characters I think. So we can make new function called `remove_quotes_expanded_arg_chars` that:
1. First checks if the word is indeed expanded, only containing the characters above (to do this we need to traverse the whole argument hierarchically.
2. Then removes the outer layers of quotes (if any).

Reply:
I attempted this in my latest commit but seems to be breaking some tests. Not sure if I am making a obvious error